tensilelite/Tensile/SolutionStructs/Solution.py

```diff
@@ -1716,21 +1737,31 @@ def calcLdsNumBytes(ldsPadA: int, LdsBlockSizePerPadA: int, ldsPadB: int, LdsBlo
         ldsAlign = int(64 / state["ProblemType"]["DataType"].numRegisters())
 
         if state["UnrollMajorLDSA"]:
-          ldsNumBytesA = (state["_DepthUA"] + ldsPadA) * state["MacroTileA"] * bpeA
+          if state["DirectToLdsA"]:
+            loadSize = (state[f"GlobalReadVectorWidthA"] * bpeA) * state["WavefrontSize"]
+            ldsNumBytesA = (state["_DepthUA"]) * state["MacroTileA"] * bpeA
+            ldsNumBytesA += (ldsNumBytesA // loadSize) * state["LdsPadA"] * bpeA
+          else:
+            ldsNumBytesA = (state["_DepthUA"] + ldsPadA) * state["MacroTileA"] * bpeA
         else:
           ldsNumBytesA = state["_DepthUA"] * (state["MacroTileA"] + ldsPadA) * bpeA
         padInterval = LdsBlockSizePerPadA
+
         if padInterval != 0:
           ldsNumBytesA = int((state["_DepthUA"] * state["MacroTileA"] * bpeA) / padInterval * (padInterval + ldsPadA * bpeA))
         ldsNumBytesAlignedA = roundUpToNearestMultiple(ldsNumBytesA, ldsAlign)
-
         # DirectToVgpr case, set 0 to lds related variables
         if state["DirectToVgprA"]:
           ldsNumBytesA = 0
           ldsNumBytesAlignedA = 0
 
         if state["UnrollMajorLDSB"]:
-          ldsNumBytesB = (state["_DepthUB"] + ldsPadB) * state["MacroTileB"] * bpeB
+          if state["DirectToLdsB"]:
```

Comment:
We should not need this extra calculation if LdsBlockSizePerPad is properly set for DTL case.

Reply:
We should only rely on LdsPad and LdsBlockSizePerPad to calculate LDS size.

Reply:
:facepalm: :facepalm: ... good point! Fixed.